pycytominer/cyto_utils/cells.py

```diff
@@ -430,46 +435,210 @@ def aggregate_compartment(
+            at once.  A larger number does not lead to faster compute times
+
+        Yields
+        ------
+        image_df : pandas.core.frame.DataFrame
+            A generator whose __next__() call returns a chunk of the compartment
+            table, where rows comprising a unique aggregation stratum are not split
+            between chunks, and thus groupby aggregations are valid
+
+        """
+
+        assert (
+            n_strata > 0
+        ), "Number of strata to pull into memory at once (n_strata) must be > 0"
+
+        # Columns of the compartment table we will obtain
+        cols = "*"
+        compartment_row1 = pd.read_sql(
+            sql=f"select {cols} from {compartment} limit 1",
+            con=self.conn,
+        )
+        typeof_str = ", ".join([f"typeof({x})" for x in compartment_row1.columns])
+        compartment_dtypes = pd.read_sql(
+            sql=f"select {typeof_str} from {compartment} limit 1",
+            con=self.conn,
+        )
+        dtype_dict = dict(
+            zip(
+                [s[7:-1] for s in compartment_dtypes.columns],  # strip typeof( )
+                compartment_dtypes.iloc[0].values,
+            )
+        )
+
+        # Obtain all valid strata combinations, and their merge_cols values
+        df_unique_mergecols = (
+            self.image_df[self.strata + self.merge_cols]
+            .groupby(self.strata)
+            .agg(lambda s: np.unique(s).tolist())
+            .reset_index(drop=True)
+        )
+
+        def _sqlite_strata_conditions(df, dtypes, n=1):
+            """Given a dataframe with columns as strata and rows as unique value
+            combinations, return a list of strings which constitute valid SQLite
+            conditional statements.
+
+            Parameters
+            ----------
+            df : pandas.core.frame.DataFrame
+                A dataframe with columns as strata and rows as unique value combos
+
+            dtypes : Dict[str, str]
+                Dictionary to look up SQLite datatype based on column name
+
+            n : int
+                Number of rows of the input df to combine in each conditional
+                statement
+
+            Returns
+            -------
+            grouped_conditions : List[str]
+                A list of strings, each string being a valid SQLite conditional
+
+            Example
+            -------
+
+                df:
+                    TableNumber | ImageNumber
+                    -------------------------
+                    [1]         | [1]
+                    [2]         | [1, 2, 3]
+                    [3]         | [1, 2]
+                    [4]         | [1]
+
+                _sqlite_strata_conditions(df,
+                                          dtypes={'TableNumber': 'integer', 'ImageNumber': 'integer'},
+                                          n=1):
+                    ['(TableNumber in (1) and ImageNumber in (1))',
```

Comment:
is this standard convention for an example? I think I'm supposed to read `_sqlite_strata_conditions(df,
                                          dtypes={'TableNumber': 'integer', 'ImageNumber': 'integer'},
                                          n=1)` as the function call and then the items right of `:` as the output. Is that correct? Is there a more intuitive way of showing what the example output might be?

Reply:
You are correct about my intent.  I was not sure what is the standard convention for an example.  Maybe I should make it more standard.